src/composables/useTokenClient.ts

```diff
@@ -97,22 +98,38 @@ export default function useTokenClient(
   const { scope = "", onError, onSuccess, ...rest } = options;
 
   const { scriptLoaded } = useGsiScript();
-  const clientId = inject<string>(GoogleClientIdKey);
+  const clientId = inject(GoogleClientIdKey);
   const isReady = ref(false);
   let client: TokenClient | undefined;
 
+  const login = (overrideConfig?: OverridableTokenClientConfig) => {
+    if (!isReady.value) {
+      if (!clientId?.value) {
```

Comment:
seems we are repeating this in multiple places, can we perhaps extract it to a composable? like `useClientId`?

Reply:
would `useClientId` be a hook intended solely for internal use within the library? If it’s not meant to be exposed externally, I think it might also make sense to modularize the validation part within the `login` function as a utility function.

Reply:
i think for now it can be used internally

yes, refactoring validation part would be nice too

Reply:
completed the refactoring.
please review the code, and let me know if any further adjustments are needed 😀